crates/aptos-rest-client/src/client_builder.rs

```diff
@@ -51,8 +51,17 @@ impl ClientBuilder {
             HeaderValue::from_static(X_APTOS_SDK_HEADER_VALUE),
         );
 
+        let mut builder = ReqwestClient::builder();
+
+        match ::std::env::var("APTOS_NO_SNI") {
+            Ok(_) => {
+                builder = builder.tls_sni(false);
+            },
+            Err(_) => ()
+        }
```

Comment:
Probably would be better to provide as an input rather than just an environment variable.  Add a function similar to `base_url` and `timeout` and you can set SNI to be turned off in the builder way which matches the design pattern currently laid out.

Reply:
At that pattern, How could I apply on Aptos CLI to turn off the SNI? through the RestClient parameter? I could only think this is only the suitable way for turn off the SNI. But yea, I should add an option to switch SNI on this impl.

Reply:
Yeah, it would be through the RestClient parameter.  Similar to this PR https://github.com/aptos-labs/aptos-core/pull/10631

Reply:
This time should be able to merge I think. Success to pass the compile now I think.